exchanges/coinbasepro/coinbasepro_test.go

```diff
@@ -2,679 +2,1453 @@ package coinbasepro
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, c, canManipulateRealOrders)
+	pID := getINTXPortfolio(t)
+	err = c.AllocatePortfolio(context.Background(), pID, testCrypto.String(), "USD", 0.001337)
+	assert.NoError(t, err)
+}
+
+func TestGetPerpetualsPortfolioSummary(t *testing.T) {
+	_, err := c.GetPerpetualsPortfolioSummary(context.Background(), "")
+	assert.ErrorIs(t, err, errPortfolioIDEmpty)
+	pID := getINTXPortfolio(t)
+	resp, err := c.GetPerpetualsPortfolioSummary(context.Background(), pID)
+	assert.NoError(t, err)
+	assert.NotEmpty(t, resp, errExpectedNonEmpty)
+}
+
+func TestGetAllPerpetualsPositions(t *testing.T) {
+	_, err := c.GetAllPerpetualsPositions(context.Background(), "")
+	assert.ErrorIs(t, err, errPortfolioIDEmpty)
+	pID := getINTXPortfolio(t)
+	_, err = c.GetAllPerpetualsPositions(context.Background(), pID)
+	assert.NoError(t, err)
+}
+
+func TestGetPerpetualsPositionByID(t *testing.T) {
+	_, err := c.GetPerpetualsPositionByID(context.Background(), "", "")
+	assert.ErrorIs(t, err, errPortfolioIDEmpty)
+	_, err = c.GetPerpetualsPositionByID(context.Background(), "meow", "")
+	assert.ErrorIs(t, err, errProductIDEmpty)
+	pID := getINTXPortfolio(t)
+	_, err = c.GetPerpetualsPositionByID(context.Background(), pID, testPair.String())
+	assert.NoError(t, err)
+}
+
+func TestGetTransactionSummary(t *testing.T) {
+	_, err := c.GetTransactionSummary(context.Background(), time.Unix(2, 2), time.Unix(1, 1), "", "", "")
+	assert.ErrorIs(t, err, common.ErrStartAfterEnd)
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, c)
+	resp, err := c.GetTransactionSummary(context.Background(), time.Unix(1, 1), time.Now(), testFiat.String(),
+		asset.Spot.Upper(), "UNKNOWN_CONTRACT_EXPIRY_TYPE")
+	assert.NoError(t, err)
+	assert.NotEmpty(t, resp, errExpectedNonEmpty)
+}
+
+func TestCreateConvertQuote(t *testing.T) {
+	_, err := c.CreateConvertQuote(context.Background(), "", "", "", "", 0)
+	assert.ErrorIs(t, err, errAccountIDEmpty)
+	_, err = c.CreateConvertQuote(context.Background(), "meow", "123", "", "", 0)
+	assert.ErrorIs(t, err, errAmountEmpty)
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, c, canManipulateRealOrders)
+	fromAccID, toAccID := convertTestHelper(t)
+	resp, err := c.CreateConvertQuote(context.Background(), fromAccID, toAccID, "", "", 0.01)
+	assert.NoError(t, err)
+	assert.NotEmpty(t, resp, errExpectedNonEmpty)
+}
+
+func TestCommitConvertTrade(t *testing.T) {
+	convertTestShared(t, c.CommitConvertTrade)
+}
+
+func TestGetConvertTradeByID(t *testing.T) {
+	convertTestShared(t, c.GetConvertTradeByID)
+}
+
+func TestGetV3Time(t *testing.T) {
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, c)
+	testGetNoArgs(t, c.GetV3Time)
+}
+
+func TestGetAllPaymentMethods(t *testing.T) {
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, c)
+	_, err := c.GetAllPaymentMethods(context.Background())
+	// This endpoint appears to currently be broken for all users
+	if err.Error() != errInternalMessageNil {
+		t.Error(err)
```

Comment:
You are making some assumptions that an `err` will be returned. You should be checking if the `err != nil`. Since I did not receive an error, this test panics:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x18 pc=0x104889870]

goroutine 259 [running]:
testing.tRunner.func1.2({0x104a44900, 0x104e518d0})
	/opt/homebrew/Cellar/go/1.22.1/libexec/src/testing/testing.go:1631 +0x1c4
testing.tRunner.func1()
	/opt/homebrew/Cellar/go/1.22.1/libexec/src/testing/testing.go:1634 +0x33c
panic({0x104a44900?, 0x104e518d0?})
	/opt/homebrew/Cellar/go/1.22.1/libexec/src/runtime/panic.go:770 +0x124
github.com/thrasher-corp/gocryptotrader/exchanges/coinbasepro.TestGetAllPaymentMethods(0x140003a6820)
	/Users/WOW/go/src/github.com/thrasher-corp/gocryptotrader/exchanges/coinbasepro/coinbasepro_test.go:601 +0x60
testing.tRunner(0x140003a6820, 0x104af7e68)
	/opt/homebrew/Cellar/go/1.22.1/libexec/src/testing/testing.go:1689 +0xec
created by testing.(*T).Run in goroutine 1
	/opt/homebrew/Cellar/go/1.22.1/libexec/src/testing/testing.go:1742 +0x318
```

Reply:
So please update your tests to check for `nil` errors before calling `err.Error()`. However, along with my other commentary about changing the error strings to `errors.New` it will be much easier to handle these issues and you won't need to call `err.Error()`. Other exchanges like `binance_test.go` will be helpful in that regard such as `TestWrapperGetServerTime`

Reply:
Yeah that was just me implementing a dumb way of being notified when that endpoint stopped returning an error for everyone. Fixed, and I'll try to avoid that sort of thing in the future.